DawnSeekersUnity/Assets/Map/Scripts/MapInteractionManager.cs

```diff
@@ -97,9 +95,12 @@ void MapClicked()
         }
 
         // Select the tile
-        if (PluginController.Instance.WorldState.UI.Selection.Intent == Intent.NONE)
+        if (
+            GameStateMediator.Instance.gameState.Selected.Intent == null
+            || GameStateMediator.Instance.gameState.Selected.Intent == Intent.NONE
+        )
```

Comment:
Is this guard what is preventing selecting tiles when in the construct intent?

If the guard there so that it doesn't conflict with the path selection logic, then maybe it should be:

```suggestion
        if (GameStateMediator.Instance.gameState.Selected.Intent != Intent.MOVE)
```